src/main/java/org/broadinstitute/hellbender/engine/filters/ReadFilterLibrary.java

```diff
@@ -14,73 +14,97 @@
             return read.getReadGroup() != null;}}
 
-    @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY)
+    /**
+     * Filter out unmapped reads. Umapped reads are defined by three criteria:
+     *
+     * <ul>
+     *     <li>SAM flag value 0x4</li>
+     *     <li>An asterisk for reference name or RNAME (column 3 of a SAM record)</li>
+     *     <li>A zero value for leftmost mapping position for POS (column 4 of SAM record)</li>
+     * </ul>
+     */
+    @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY, summary = "Filtr out unmapped reads")
     public static class MappedReadFilter extends ReadFilter {
         private static final long serialVersionUID = 1L;
         @Override public boolean test(final GATKRead read) {
             return !read.isUnmapped();}}
 
-    @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY)
+    /** Filter out reads without available mapping quality (MAPQ=255). */
+    @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY, summary = "Filter out reads without available mapping quality")
     public static class MappingQualityAvailableReadFilter extends ReadFilter {
         private static final long serialVersionUID = 1L;
         @Override public boolean test(final GATKRead read) {
             return read.getMappingQuality() != QualityUtils.MAPPING_QUALITY_UNAVAILABLE;}}
 
-    @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY)
+    /** Filter out reads with mapping quality equal to zero. */
+    @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY, summary = "Filter out reads with mapping quality equal to zero")
     public static class MappingQualityNotZeroReadFilter extends ReadFilter {
         private static final long serialVersionUID = 1L;
         @Override public boolean test(final GATKRead read) {
             return read.getMappingQuality() != 0;}}
 
     /**
-     * Reads that either have a mate that maps to the same contig, or don't have a mapped mate.
+     * Keep reads that have a mate that maps to the same contig (RNEXT is "="), is single ended (not 0x1) or has an unmapped mate (0x8).
+     *
+     * See MappedReadFilter for criteria defining an unmapped read.
      */
-    @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY)
+    @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY, summary = "Keep reads that have a mate mapping to the same contig or it is unmapped", extraDocs = MappedReadFilter.class)
```

Comment:
The read filters that require a positive test result should say "Keep only reads that have X" to make it clear what is the behavior -- otherwise it's ambiguous.

(this applies to all others that currently say "Keep reads that X")